monai/transforms/utils.py

```diff
@@ -2051,5 +2058,136 @@ def has_status_keys(data: torch.Tensor, status_key: Any, default_message: str =
+
+    For details about the implementation, check out the
+    `SciPy<https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.distance_transform_edt.html/>`_
+    and `cuCIM<https://docs.rapids.ai/api/cucim/nightly/api/#cucim.core.operations.morphology.distance_transform_edt/>`_ docs.
+
+    Args:
+        img: Input image on which the distance transform shall be run.
+            channel first array, must have shape: (num_channels, H[, W, ..., ])
+            Input gets passed channel-wise to the distance-transform, thus results from this function may differ
+            from directly calling ``distance_transform_edt()`` in CuPy or scipy.
+        sampling: Spacing of elements along each dimension. If a sequence, must be of length equal to the input rank;
+            if a single number, this is used for all axes. If not specified, a grid spacing of unity is implied.
+        return_distances: Whether to calculate the distance transform.
+        return_indices: Whether to calculate the feature transform.
+        distances: An output array to store the calculated distance transform, instead of returning it.
+            `return_distances` must be True.
+        indices: An output array to store the calculated feature transform, instead of returning it. `return_indicies` must be True.
+        block_params: This parameter is specific to cuCIM and does not exist in SciPy. For details, look here
+            `distance_transform_edt<https://docs.rapids.ai/api/cucim/nightly/api/#cucim.core.operations.morphology.distance_transform_edt/>`_
+        float64_distances: This parameter is specific to cuCIM and does not exist in SciPy.
+            If True, use double precision in the distance computation (to match SciPy behavior).
+            Otherwise, single precision will be used for efficiency.
+
+    Returns:
+        distances: The calculated distance transform. Returned only when `return_distances` is True and `distances` is not supplied.
+            It will have the same shape as image. For cuCIM: Will have dtype torch.float64 if float64_distances is True,
+            otherwise it will have dtype torch.float32. For scipy: Will have dtype np.float64.
+        indices: The calculated feature transform. It has an image-shaped array for each dimension of the image.
+            Returned only when `return_indices` is True and `indices` is not supplied. dtype np.float64.
+
+    """
+    distance_transform_edt, has_cucim = optional_import(
+        "cucim.core.operations.morphology", name="distance_transform_edt"
+    )
+    use_cp = has_cp and has_cucim and isinstance(img, torch.Tensor) and img.device.type == torch.device("cuda").type
+
+    if not return_distances and not return_indices:
+        raise RuntimeError("Neither return_distances nor return_indices True")
+    distances_original, indices_original = distances, indices
+    distances, indices = None, None
```

Comment:
Maybe we should also check if ndims is in the expected range (so I guess between 2 and 4 dims for 1D to 3D)

Reply:
I have added that now. As above, cupy does not support 1D data, so it has to be `if not (img.ndim >= 3 and img.ndim <= 4):`
As we discussed scipy and cupy shall behave in the same way so I guess it makes sense to raise an error there